lib/util/package-rules/repositories.ts

```diff
@@ -0,0 +1,47 @@
+import is from '@sindresorhus/is';
+import { minimatch } from 'minimatch';
+import type { PackageRule, PackageRuleInputConfig } from '../../config/types';
+import { logger } from '../../logger';
+import { regEx } from '../regex';
+import { Matcher } from './base';
+import { massagePattern } from './utils';
+
+export class RepositoriesMatcher extends Matcher {
+  override matches(
+    { repository }: PackageRuleInputConfig,
+    { matchRepositories }: PackageRule
+  ): boolean | null {
+    if (is.undefined(matchRepositories)) {
+      return null;
+    }
+
+    if (is.undefined(repository)) {
+      return false;
+    }
+
+    return matchRepositories.some(
+      (matchRepositoryName) =>
+        isRegexAndMatches(matchRepositoryName, repository) ||
+        minimatch(repository, matchRepositoryName, { dot: true })
+    );
+  }
+}
+
+function isRegexAndMatches(repoPattern: string, repo: string): boolean {
+  if (
+    repoPattern.length < 2 ||
+    !repoPattern.startsWith('/') ||
+    !repoPattern.endsWith('/')
+  ) {
+    return false;
+  }
+
+  const repoPatternWithoutSlashes = repoPattern.slice(1, -1);
+
+  const re = regEx(massagePattern(repoPatternWithoutSlashes));
```

Comment:
I think this will throw an error if it's an invalid regex, right?

Should we catch this during config validation instead of runtime?

Reply:
Good callout. I think a few of the other package-rules are susceptible to this currently... 

I've added handling for invalid regexes